lib/Runtime/Base/FunctionBody.h

```diff
@@ -1348,16 +1393,180 @@ namespace Js
         ScriptFunctionType * AllocDeferredPrototypeType();
     };
 
+    inline Js::LocalFunctionId FunctionProxy::GetLocalFunctionId() const
+    {
+        Assert(GetFunctionInfo());
+        return GetFunctionInfo()->GetLocalFunctionId();
+    }
+
+    inline void FunctionProxy::SetLocalFunctionId(LocalFunctionId functionId)
+    {
+        Assert(GetFunctionInfo());
+        Assert(GetFunctionInfo()->GetFunctionProxy() == this);
```

Comment:
There are cases where it’s awkward to assert in GetFunctionInfo() because we’re in the midst of swapping the functionBodyImpl. I suppose we could have a GetFunctionInfoUnchecked() for those.

--Paul

From: Hitesh Kanwathirtha [mailto:notifications@github.com]
Sent: Thursday, July 21, 2016 11:04 AM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Paul Leathers <pleath@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Microsoft/ChakraCore] Restructure FunctionBody hierarchy for re-deferral (#1278)


In lib/Runtime/Base/FunctionBody.h<https://github.com/Microsoft/ChakraCore/pull/1278#discussion_r71755941>:

> @@ -1348,16 +1393,180 @@ namespace Js

>          ScriptFunctionType * AllocDeferredPrototypeType();

>      };

>

> +    inline Js::LocalFunctionId FunctionProxy::GetLocalFunctionId() const

> +    {

> +        Assert(GetFunctionInfo());

> +        return GetFunctionInfo()->GetLocalFunctionId();

> +    }

> +

> +    inline void FunctionProxy::SetLocalFunctionId(LocalFunctionId functionId)

> +    {

> +        Assert(GetFunctionInfo());

> +        Assert(GetFunctionInfo()->GetFunctionProxy() == this);

Can these asserts be moved to GetFunctionInfo itself? Also was curious why this particular assert wasn't there in GetLocalFunctionId (another reason I suggested centralizing the asserts)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/1278/files/b84eb2bf59a60a0647d0efcece9bc741d4855b53#r71755941>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APF8RLQXTH-jjUOD-IA2WsNpgLf4uipiks5qX7SOgaJpZM4JM3EN>.